chia/cmds/plotnft_funcs.py

```diff
@@ -270,106 +274,133 @@ async def submit_tx_with_confirmation(
         print(f"Error performing operation on Plot NFT -f {fingerprint} wallet id: {wallet_id}: {e}")
 
 
+async def wallet_id_lookup_and_check(wallet_client: WalletRpcClient, wallet_id: Optional[int]) -> int:
+    selected_wallet_id: int
+    try:
+        pool_wallets = await wallet_client.get_wallets(wallet_type=WalletType.POOLING_WALLET)
+
+        if wallet_id is None:
+            if len(pool_wallets) == 0:
+                raise CliRpcConnectionError(
+                    "No pool wallet found. Use 'chia plotnft create' to create a new pooling wallet."
+                )
+            if len(pool_wallets) > 1:
+                raise CliRpcConnectionError("More than one pool wallet found. Use -i to specify pool wallet id.")
+            selected_wallet_id = pool_wallets[0]["id"]
+        else:
+            selected_wallet_id = wallet_id
+
+        if not any(wallet["id"] == selected_wallet_id for wallet in pool_wallets):
+            raise CliRpcConnectionError(f"Wallet with id: {selected_wallet_id} is not a pool wallet.")
+
+        return selected_wallet_id
+    except ValueError as e:
+        raise CliRpcConnectionError(f"Error: {type(e).__name__}: {e}")
```

Comment:
Does this really need to cover the entire `try:` block?  It feels like this is perhaps focused on `wallet_client.get_wallets()`?  Though the client raises a specific error now on `"success": false` so maybe I'm looking in the wrong place.

Reply:
I'm not opposed to changing this code - but this was how it was originally

Reply:
oh, sorry my mistake - yea, this is new code duh. yea, the try block really only needs to be around `get_wallets`

Reply:
Updated